packages/mobile/src/components/comments/CommentSection.tsx

```diff
@@ -106,12 +110,21 @@ const CommentSectionContent = () => {
 
 type CommentSectionProps = {
   entityId: ID
-  isDrawerOpen: boolean
-  setIsDrawerOpen: (isOpen: boolean) => void
 }
 
 export const CommentSection = (props: CommentSectionProps) => {
-  const { entityId, isDrawerOpen, setIsDrawerOpen } = props
+  const { entityId } = props
+
+  const { params } = useRoute<'Track'>()
+  const { showComments } = params ?? {}
+
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false)
+
+  useEffectOnce(() => {
+    if (showComments) {
+      setIsDrawerOpen(true)
+    }
+  })
```

Comment:
Am I crazy or did you just change this in the other PR? Whats the motivation for moving it back

Reply:
Kind of, before the state was tracked in redux in the Drawers slice. I had to change that to fix the navigation stuff. But keeping the state in TrackScreen caused the whole screen to rerender when the drawer opened. Now only the small comment section needs to rerender!